src/model/transaction/setBlockDataForContentState.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule setBlockDataForContentState
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+var Immutable = require('immutable');
+
+import type ContentState from 'ContentState';
+import type SelectionState from 'SelectionState';
+
+function setBlockDataForContentState(
```

Comment:
both scenario are valid. In my case lines have icons at the beginning. Icons that should not not be propagated.

It would be nice to be able to tweak that kind of behaviors easily.